File: Sum-of-Square-Numbers/SoSN.py
import math


class Solution:
    def judgeSquareSum(self, c):
        """
        :type c: int
        :rtype: bool
        """

        if c == 0:
            return True
        elif c < 0:
            return False

        for i in range(0, int(math.sqrt(c))+1):
            tempI = i ** 2
            try:
                tempRest = math.sqrt(c-tempI)
            except:
                continue
            if int(tempRest) == tempRest:
                return True

        return False


a = Solution()
a.judgeSquareSum(5)

File: Sum-of-Square-Numbers/SoSN.rs
fn judge_square_sum(input: i32) -> bool {
    if input == 0 {
        return true;
    }

    if input < 0 {
        return false;
    }

    let input_cnvrt = input as f32;

    for i in 0..input_cnvrt.sqrt() as i32 + 1 {
        let temp_i = i.pow(2);
        let mut temp_rest = (input - temp_i) as f32;
        temp_rest = temp_rest.sqrt();
        //println!("{}", temp_rest);
        if temp_rest == temp_rest.round() {
            return true;
        }
    }

    false
}
fn main() {
    println!("{}", judge_square_sum(0));
    println!("{}", judge_square_sum(5));
    println!("{}", judge_square_sum(2));

    println!("{}", judge_square_sum(3));
}
